invenio_app_ils/ui/backoffice/src/pages/UserDetails/state/__tests__/reducer.js

```diff
@@ -0,0 +1,45 @@
+import reducer, { initialState } from '../reducer';
+import * as types from '../types';
+
+describe('Fetch user details reducer', () => {
+  it('should have initial state', () => {
+    expect(reducer(undefined, {})).toEqual(initialState);
+  });
+
+  it('should change loading state on loading action', () => {
+    const action = {
+      type: types.IS_LOADING,
+    };
+    expect(reducer(initialState, action)).toEqual({
+      ...initialState,
+      isLoading: true,
+    });
+  });
+
+  it('should change data state on success action', () => {
+    const user = { id: '1' };
+    const action = {
+      type: types.SUCCESS,
+      payload: user,
+    };
+    expect(reducer(initialState, action)).toEqual({
+      ...initialState,
+      isLoading: false,
+      data: user,
+      hasError: false,
+    });
+  });
+
+  it('should change error state on error action', () => {
+    const action = {
+      type: types.HAS_ERROR,
+      payload: 'Error',
+    };
+    expect(reducer(initialState, action)).toEqual({
+      ...initialState,
+      isLoading: false,
+      data: 'Error',
```

Comment:
I wouldn't mix the data type to be both an object or string, just for consistency and to know what we expect.